server.js

```diff
@@ -1,49 +1,43 @@
-let categories = require('./my-app-react/src/data/categories.json')
-let products = require('./my-app-react/src/data/products.json')
+const dataService = require('./dataservice.js');
+const fromEntries = require('object.fromentries');
 
-const express = require('express')
-const app = express()
+const express = require('express');
+const app = express();
 
 app.get('/info', (req, res) => {
-    info = [
-        {"serverName": "localhost5000"},
-        {"sampleName": "chinemerem-react"},
-        {"serverVersion": "1.0.0"}
-    ]
+    const serverInfo = require('package')
+    const info = [
+        {"serverName": serverInfo.name},
+        {"serverVersion": serverInfo.version}
+    ];
     res.json(info)
-})
+});
 
 app.get('/products/all', (req, res) => {
-    res.json(products)
-})
+    const productsMap = dataService.getCombinedProductMap();
+    res.send( fromEntries(productsMap));
+});
 
 app.get('/products/:id', (req, res) => {
-    let urlProductID = req.params.id
-    let productsLocal = products.products
-
-    for (productID in productsLocal) {
-        let product = productsLocal[productID];
-        if(urlProductID == product.id) {
-            res.json(product)
-        }
-    }
-
-})
+    const urlProductID = req.params.id;
+    const productsMap = dataService.getCombinedProductMap();
+    
+    res.send(productsMap.get(urlProductID));
+});
 
 app.get('/category/:id', (req, res) => {
-    let urlCategoryID = req.params.id
-    let productsLocal = products.products
-    let arrayOfProducts = []
+    const urlCategoryID = req.params.id;
+    const combinedProductMap = dataService.getCombinedProductMap();
+    const productsBasedOnCategory = new Map();
 
-    for (productID in productsLocal) {
-        let product = productsLocal[productID];
-        if(urlCategoryID == product.categoryId) {
-            arrayOfProducts.push(product)
+    combinedProductMap.forEach(((value, key) => {
+        if (value.categoryId === urlCategoryID) {
+            productsBasedOnCategory.set(key, value);
         }
-    }
+    }));
     
-    res.json(arrayOfProducts)
-})
+    res.send(fromEntries(productsBasedOnCategory));
```

Comment:
```suggestion
    res.send(Object.fromEntries(productsBasedOnCategory));
```

Reply:
Please is this all I need to do to complete this task, cos I've been on this for 11 days. That is really time-consuming for just one task. I'd like to do all the needful and submit the task once and for all. Thanks.

Reply:
@felixfavour You haven't address all our comments, so you're the only person to blame. Here's [the docs for that method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries).